### PR TITLE
fix TestDealPublisher

### PR DESCRIPTION
**Pull request by Stebalien**
This test _should_ be mostly deterministic, but is still failing. I _believe_ this is because:

1. We record the "start" time before starting the timer.
2. In the test, we increment the clock till right after the timer should fire, if the timer start time matches the recorded start time.
3. If we happen to advance the mock clock between recording the time and starting the timer, the timer will never fire.

Unfortunately, I can't reproduce this locally (at all) so I'm mostly just guessing.

### CODE REVIEW

File: markets/storageadapter/dealpublisher.go [hunk 1]
```diff
@@ -384,12 +388,12 @@ func pieceCids(deals []market2.ClientDealProposal) string {
 
 // filter out deals that have been cancelled
 func (p *DealPublisher) filterCancelledDeals() {
-	i := 0
+	filtered := p.pending[:0]
```
Comment (Stebalien) [inline markets/storageadapter/dealpublisher.go:391]: This was just a drive-by fix to make the logic here easier to reason about.

File: markets/storageadapter/dealpublisher.go [hunk 2]
```diff
@@ -257,7 +261,7 @@ func (p *DealPublisher) publishAllDeals() {
 
 	// Filter out any deals that have been cancelled
 	p.filterCancelledDeals()
-	deals := p.pending[:]
+	deals := p.pending
```
Comment (Stebalien) [inline markets/storageadapter/dealpublisher.go:264]: Also a drive-by fix. There's no need to do this unless `p.pending` is an array.

**Comment by Stebalien** [comment]
codecove -> :facepalm:

**Comment by Stebalien** [comment]
Flaky test disabled in https://github.com/filecoin-project/lotus/pull/7176.

**Review by arajasek — APPROVED** [review]
